VideoLocker/src/main/java/org/edx/mobile/view/CourseDashboardFragment.java

```diff
@@ -159,6 +173,56 @@ public void onActivityCreated(Bundle savedInstanceState) {
         courseTextName.setText(courseData.getCourse().getName());
         CourseEntry course = courseData.getCourse();
         courseTextDetails.setText(course.getDescription(getActivity(), true));
+
+        if (environment.getConfig().isShareCourseEnabled()) {
+            shareButton.setVisibility(headerImageView.VISIBLE);
+            shareButton.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    openShareMenu();
+                }
+            });
+        }
+    }
+
+    /**
+     * Creates a dropdown menu with appropriate apps when the share button is clicked.
+     */
+    private void openShareMenu() {
+        ShareUtils.showShareMenu(
+                ShareUtils.newShareIntent(courseData.getCertificateURL(), null),
+                getActivity().findViewById(R.id.course_detail_share),
+                new ShareUtils.ShareMenuItemListener() {
+                    @Override
+                    public void onMenuItemClick(@NonNull ComponentName componentName) {
+                        String shareText = ResourceUtil.getFormattedString(
+                                getResources(),
+                                R.string.share_course,
+                                "platform_name",
+                                getString(R.string.platform_name)).toString();
+                        String shareLink = courseData.getCourse().getCourse_about();
+                        String event_type = componentName.getPackageName().toString();
+                        System.out.println("Share course event:" + event_type);
+                        switch (event_type) {
```

Comment:
@BenjiLee If you pull my changes from master, you can just pass the ComponentName to ISegment and use ISegmentImpl.getShareTypeFromComponentName(componentName)